### ISSUE TITLE
get more details from Hikari for LS2.x

### ISSUE THREAD

**Issue by ctramnitz** [theangryangel/logstash-output-jdbc#79]
I'm trying to use jdbc ouput with Hive (HiveServer2). Since we have LS2 I'm using 0.3.2 of the plugin. I can see that required jar's are being loading, but then I get a PoolInitializationException.
There were a few issues in Hive that could have caused this (such as isValid), but as far as I can tell, they should be fixed in my version (2.1.0 from HDP 2.5).

I don't think this a logstash-output-jdbc or Hive problem but a rather a configuration or dependency issue, but with the limited output from Hikari I can't tell whats wrong.
Config:
```
 jdbc {
                driver_class => "org.apache.hive.jdbc.HiveDriver"
                username => "hive"
                connection_string => "jdbc:hive2://server:10000/db"
                statement => [ "UPSERT INTO jdbctest (host, ts, message) VALUES (?, ?, ?)", "host", "@timestamp", "message" ]
            }
```
Output:
```
# /opt/logstash/bin/logstash -f /etc/logstash/conf.d/logstash.conf --debug
[...]
JDBC - Starting up {:level=>:info, :file=>"logstash/outputs/jdbc.rb", :line=>"99", :method=>"register"}
log4j java properties setup {:log4j_level=>"DEBUG", :level=>:debug, :file=>"logstash/logging.rb", :line=>"89", :method=>"setup_log4j"}
JDBC - jarpath {:path=>"/opt/logstash/vendor/jar/jdbc/*.jar", :level=>:debug, :file=>"logstash/outputs/jdbc.rb", :line=>"180", :method=>"load_jar_files!"}
JDBC - Loaded jar [many times]
Pipeline aborted due to error {:exception=>"Java::ComZaxxerHikariPool::HikariPool::PoolInitializationException", :backtrace=>["com.zaxxer.hikari.pool.HikariPool.initializeConnections(com/zaxxer/hikari/pool/HikariPool.java:535)", "com.zaxxer.hikari.pool.HikariPool.<init>(com/zaxxer/hikari/pool/HikariPool.java:137)", "com.zaxxer.hikari.HikariDataSource.getConnection(com/zaxxer/hikari/HikariDataSource.java:94)", "java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)", "RUBY.setup_and_test_pool!(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-jdbc-0.3.2-java/lib/logstash/outputs/jdbc.rb:157)", "RUBY.register(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-jdbc-0.3.2-java/lib/logstash/outputs/jdbc.rb:116)", "RUBY.register(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.0-java/lib/logstash/output_delegator.rb:75)", "RUBY.start_workers(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.0-java/lib/logstash/pipeline.rb:181)", "org.jruby.RubyArray.each(org/jruby/RubyArray.java:1613)", "RUBY.start_workers(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.0-java/lib/logstash/pipeline.rb:181)", "RUBY.run(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.0-java/lib/logstash/pipeline.rb:136)", "RUBY.start_pipeline(/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.0-java/lib/logstash/agent.rb:491)", "java.lang.Thread.run(java/lang/Thread.java:745)"], :level=>:error, :file=>"logstash/agent.rb", :line=>"493", :method=>"start_pipeline"}
```

* Version of plugin used: 0.3.2
* Version of Logstash used:  2.4.0
* Database engine & version you're connecting to: HiveServer2 2.1.0
* Have you checked you've met the Logstash requirements for Java versions?: Yes, running Oracle JDK 1.8.0u112

**Comment by theangryangel** [comment]
Wiring up Hikari's logging to logstash has been a bit of a pain in the arse in 2.x of Logstash, unfortunately. I *think* if you run 2.x with `-debug` you should get more information.

However, that said, last time this happened it was down to the JDBC driver (phoenix) and how Hikari initially sets up the pool. I provided a couple of configuration options for the 5.x line to deal with that other issue, but I never backported them. If you can wait a few hours I'll try and find the time, or I can point you at the lines you can insert directly into your installed copy to try it out?

**Comment by ctramnitz** [comment]
Thanks for the quick reply. I was already running with debug and this is all I've got.
Before you dive into the details, let me try on another box with LS5 to see if I can get any more output. If you have any hints for the JDBC config for the other issue (beyond what is in the Phoenix sample configs) and/or extended Hikari logging for LS5 that would be great.

**Comment by ctramnitz** [comment]
I've got a little further on a LS5 system now, no more error messages, but now I get a NPE.
Is log4j2.xml still the method to get debug log from Hikari? It seems LS5 uses a properties file rather than xml or does the log4j2.xml need to go into the gem path?

**Comment by theangryangel** [comment]
Sorry for the delay @ctramnitz I've not had any time during the week to really dig into whats going on. 

It looks like at some point during the 5.x release I made a mistake and never upgraded the log4j-slf4j bridge that HikariCP needs to integrate with the rest of the logging under LS5 properly.

I've just pushed logstash-output-jdbc-5.2.0 to rubygems.org. If you upgrade to that, and then you'll find with the `--debug` option that that HikariCP is **very** spammy/hopefully more useful :)

![hikaricp-debug-fix](https://cloud.githubusercontent.com/assets/309967/24578268/c9d7100c-16d4-11e7-95b5-7419a97c639f.gif)
